File: VB/WinSolution.Module.Win/CustomizeBarItemSettingsListViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Templates
Imports DevExpress.ExpressApp.Win.Templates.ActionContainers
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraBars
Imports DevExpress.Data.Filtering

Namespace WinSolution.Module.Win
	Public Class CustomizeBarItemSettingsListViewController
		Inherits ViewController
		Private parametrizedAction1 As DevExpress.ExpressApp.Actions.ParametrizedAction
		Public Sub New()
			Me.TargetViewType = ViewType.ListView
			Me.TargetObjectType = GetType(DomainObject1)
			Me.parametrizedAction1 = New DevExpress.ExpressApp.Actions.ParametrizedAction(Me, "MyDateFilter", PredefinedCategory.Search, GetType(DateTime))
			Me.parametrizedAction1.Caption = "My Date Filter"
			AddHandler parametrizedAction1.Execute, AddressOf parametrizedAction1_Execute
		End Sub
		Private Sub parametrizedAction1_Execute(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.Actions.ParametrizedActionExecuteEventArgs)
			CType(View, ListView).CollectionSource.Criteria(Me.parametrizedAction1.Id) = CriteriaOperator.Parse("[CreatedOn]=?", Convert.ToDateTime(e.ParameterCurrentValue))
		End Sub
		Protected Overrides Sub OnFrameAssigned()
			MyBase.OnFrameAssigned()
			AddHandler Frame.TemplateChanged, AddressOf Frame_TemplateChanged
		End Sub
		Private Sub Frame_TemplateChanged(ByVal sender As Object, ByVal e As EventArgs)
			For Each container As IActionContainer In Frame.Template.GetContainers()
				If container.ContainerId = PredefinedCategory.Search.ToString() AndAlso TypeOf container Is ActionContainerBarItem Then
					Dim actionContainerBarItem As ActionContainerBarItem = CType(container, ActionContainerBarItem)
					For Each itemLink As BarItemLink In actionContainerBarItem.ItemLinks
						If itemLink.Caption = parametrizedAction1.Caption Then
							Dim editItemLink As BarEditItemLink = CType(itemLink, BarEditItemLink)
							Dim editor As RepositoryItemDateEdit = CType(editItemLink.Edit, RepositoryItemDateEdit)
							editor.Mask.UseMaskAsDisplayFormat = True
							editor.Mask.EditMask = "MM/dd/yyyy"
							Exit For
						End If
					Next itemLink
				End If
			Next container
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module.Win/CustomizeBarItemSettingsListViewController.cs
using System;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Templates;
using DevExpress.ExpressApp.Win.Templates.ActionContainers;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraBars;
using DevExpress.Data.Filtering;

namespace WinSolution.Module.Win {
    public class CustomizeBarItemSettingsListViewController : ViewController {
        private DevExpress.ExpressApp.Actions.ParametrizedAction parametrizedAction1;
        public CustomizeBarItemSettingsListViewController() {
            this.TargetViewType = ViewType.ListView;
            this.TargetObjectType = typeof(DomainObject1);
            this.parametrizedAction1 = new DevExpress.ExpressApp.Actions.ParametrizedAction(this, "MyDateFilter", PredefinedCategory.Search, typeof(DateTime));
            this.parametrizedAction1.Caption = "My Date Filter";
            this.parametrizedAction1.Execute += new DevExpress.ExpressApp.Actions.ParametrizedActionExecuteEventHandler(parametrizedAction1_Execute);
        }
        void parametrizedAction1_Execute(object sender, DevExpress.ExpressApp.Actions.ParametrizedActionExecuteEventArgs e) {
            ((ListView)View).CollectionSource.Criteria[this.parametrizedAction1.Id] = CriteriaOperator.Parse("[CreatedOn]=?", Convert.ToDateTime(e.ParameterCurrentValue));
        }
        protected override void OnFrameAssigned() {
            base.OnFrameAssigned();
            Frame.TemplateChanged += new EventHandler(Frame_TemplateChanged);
        }
        void Frame_TemplateChanged(object sender, EventArgs e) {
            foreach (IActionContainer container in Frame.Template.GetContainers()) {
                if (container.ContainerId == PredefinedCategory.Search.ToString() && container is ActionContainerBarItem) {
                    ActionContainerBarItem actionContainerBarItem = (ActionContainerBarItem)container;
                    foreach (BarItemLink itemLink in actionContainerBarItem.ItemLinks) {
                        if (itemLink.Caption == parametrizedAction1.Caption) {
                            BarEditItemLink editItemLink = (BarEditItemLink)itemLink;
                            RepositoryItemDateEdit editor = (RepositoryItemDateEdit)editItemLink.Edit;
                            editor.Mask.UseMaskAsDisplayFormat = true;
                            editor.Mask.EditMask = "MM/dd/yyyy";
                            break;
                        }
                    }
                }
            }
        }
    }
}